torch/csrc/distributed/c10d/init.cpp

```diff
@@ -1255,6 +1255,14 @@ that adds a prefix to each key inserted to the store.
 static const auto StoreTorchBind =
     torch::class_<::c10d::Store>("dist_c10d", "Store");
 
+static const auto FileStoreTorchBind =
+    torch::class_<::c10d::FileStore>("dist_c10d", "FileStore")
+        .def(torch::init([](const std::string& path,
+                            int64_t num_workers) {
+          return c10::make_intrusive<::c10d::FileStore>(
+              path, num_workers);
+        }));
+
```

Comment:
Can we add some unit tests verifying this works as expected in TorchScript? Also, I just noticed that users would have to use something like `torch.classes.dist_c10d.Filestore` to use this in TorchScript. Is it possible to bind `torch.distributed.FileStore` to this class instead when we are under TorchScript?

Reply:
Yes, the future plan is to use TorchBind (instead of pybind11) to bind the custom class like `FileStore` directly, but currently there're bunch of improvements need to be done between we fully switch to it. Right now this binding co-exist with pybind bindings, and unfortunately user have to use this initially as they are not convertible between each other. We will do a full switch to this binding once we finish the improvements to TorchBind, after that user can automatically use the same API like `torch.distributed.FileStore` inside TorchScript.